### PR TITLE
Towards 1.0 release

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [x] - Add missing license headers
- [x] - Update README, move details to Jenkins Wiki
- [x] - Update index.jelly
- [x] - Links to Wiki
- [x] - Move credentials stuff to "credentials" package
- [x] - Move DockerTool to "tools" package
- [x] - Move lib resources and taglib to a lib folder

@reviewbybees

### CODE REVIEW

**Comment by KostyaSha** [comment]
[ ] - Move credentials into credentials package to not trash root package.

**Comment by oleg-nenashev** [comment]
@KostyaSha agreed

**Comment by oleg-nenashev** [comment]
Fixed the comments
@reviewbybees @KostyaSha @rsandell @jtnord

File: README.md [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
 
-This allows users to configure one set of endpoint/credentials and use it across all the Docker
-related plugins, thereby keeping configuration more [DRY](http://en.wikipedia.org/wiki/Don%27t_repeat_yourself).
 
-See [here](src/test/java/org/jenkinsci/plugins/docker/commons/SampleDockerBuilder.java)
-and [here](src/test/java/org/jenkinsci/plugins/docker/commons/SampleDockerRegistryBuilder.java)
-and their corresponding `config.jelly` files for an example of how to use them.
+License
+---
+[MIT License](http://opensource.org/licenses/MIT)
```
Comment (jtnord) [inline README.md]: This should be in LICENCE.txt (which is [recommended](https://help.github.com/articles/open-source-licensing/#where-does-the-license-live-on-my-repository) by github)

**Comment by rsandell** [comment]
A LICENSE.txt in the root or spell it out fully in the readme.

**Comment by KostyaSha** [comment]
imho no sense to pollute every file with license header that never will be updated right with mentioning all contributors to this files. It should be enough to have only one license file in repository root.

**Comment by KostyaSha** [comment]
Checkstyle/pmd rules that will enforce tabs or spaces will be also good idea and reduce efforts on commenting future PRs with "please remove whitespace changes because current code random-styled".

**Comment by oleg-nenashev** [comment]
@KostyaSha I'm going to kick-off the last "beta" release. After that it makes sense to add such checks (it's not a **must** btw)

File: src/test/resources/org/jenkinsci/plugins/docker/commons/util/SampleDockerBuilder/config.jelly [hunk 2]
```diff
@@ -27,5 +27,5 @@ THE SOFTWARE.
 <j:jelly xmlns:j="jelly:core" xmlns:f="/lib/form">
     <f:property field="server"/>
     <f:property field="registry"/>
-    <d:selectDockerTool field="toolName" xmlns:d="/org/jenkinsci/plugins/docker/commons"/>
+    <d:selectDockerTool field="toolName" xmlns:d="/org/jenkinsci/plugins/docker/commons/lib"/>
```
Comment (jtnord) [inline src/test/resources/org/jenkinsci/plugins/docker/commons/util/SampleDockerBuilder/config.jelly]: file is modified update (c)

**Comment by KostyaSha** [comment]
> @KostyaSha I'm going to kick-off the last "beta" release. After that it makes sense to add such checks (it's not a must btw)

Everybody complain about coding style in core, but when you have chance to do new plugin and exclude this future headache you not doing this. This not understandable for me.

**Comment by jtnord** [comment]
lets do the code refactor in a separate PR.
Now who wants to start a tabs vs space war :-)

**Comment by jtnord** [comment]
@oleg-nenashev are you doing follow up PRs for affected plugins?

**Comment by KostyaSha** [comment]
@jtnord tabs enforcement doesn't work out-of-the box in checkstyle afair (i gave up configuring it), so spaces may win :)

**Comment by jtnord** [comment]
@KostyaSha its a simple RegexpSinglelineJava  instead - I reject your 4 spaces per indent and substitute and arbitrary user preference called "tab width" :-)

(honestly as long as its consistent and identified I don't mind, and eclipse/netbeans/intelliJ formatter is bonus)

**Comment by KostyaSha** [comment]
@jtnord regexp doesn't work as expected and i like tabs. But the main issue is not tabs vs spaces, it's mix of both that can be avoid right now ;)

**Comment by oleg-nenashev** [comment]
> @oleg-nenashev are you doing follow up PRs for affected plugins?

Yes, somebody will have to do it

**Comment by tfennelly** [comment]
:+1: from what I can see (mostly copyright headers + package moving)

**Comment by jtnord** [comment]
:+1: from me

**Comment by KostyaSha** [comment]
@oleg-nenashev forgot move impl folder?

**Comment by KostyaSha** [comment]
@oleg-nenashev and now released?

**Comment by oleg-nenashev** [comment]
@KostyaSha 
My bad. I've written a response to your comment in one of my tabs, but seems I have not submitted it.
M statement was that it's preferable to store implementations for extension points and APIs in $(root)/impl (it's IMHO and very debatable BTW).

If you don't mind, I'll keep it as is now and create an enhancement version for 2.0 version when it happens

**Comment by KostyaSha** [comment]
I just expected `impl`s under corresponding packages...

**Pull request merged**